eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -0,0 +1,350 @@
+                FeatureCodecHeader featureCodecHeader = vcfCodec.readHeader(sourceFromStream);
+                headers.put(source.getStudyId(), (VCFHeader) featureCodecHeader.getHeaderValue());
+            }
+        }
+        
+        for (String studyId : sources.keySet()) {
+            if (!headers.containsKey(studyId)) {
+                throw new IllegalArgumentException("file headers not available for study " + studyId);
+            }
+        }
+
+        return headers;
+        //TODO: allow specify which samples to return
+/*
+//        List<String> returnedSamples = new ArrayList<>();
+//        if (options != null) {
+//            returnedSamples = options.getAsStringList(VariantDBAdaptor.VariantQueryParams.RETURNED_SAMPLES.key());
+//        }
+        int lastLineIndex = fileHeader.lastIndexOf("#CHROM");
+        if (lastLineIndex >= 0) {
+            String substring = fileHeader.substring(0, lastLineIndex);
+            if (returnedSamples.isEmpty()) {
+                BiMap<Integer, String> samplesPosition = StudyConfiguration.getSamplesPosition(studyConfiguration).inverse();
+                returnedSamples = new ArrayList<>(samplesPosition.size());
+                for (int i = 0; i < samplesPosition.size(); i++) {
+                    returnedSamples.add(samplesPosition.get(i));
+                }
+            }
+            String samples = String.join("\t", returnedSamples);
+            logger.debug("export will be done on samples: [{}]", samples);
+
+            fileHeader = substring + "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t" + samples;
+        }
+//        return VariantFileMetadataToVCFHeaderConverter.parseVcfHeader(fileHeader);
+
+        // */
+
+    }
+
+    public int getFailedVariants() {
+        return failedVariants;
+    }
+
+    class VariantFields {
+        public int start;
+        public int end;
+        public String reference;
+        public String alternate;
+    }
+
+    /**
+     * converts org.opencb.biodata.models.variant.Variant into one or more htsjdk.variant.variantcontext.VariantContext
+     * behaviour:
+     * * one VariantContext per study
+     * * split multiallelic variant will remain split.
+     * * in case a normalized INDEL has empty alleles, the original alleles in the vcf line will be used.
+     *
+     * steps:
+     * * foreach variantSourceEntry, collect genotypes in its study, only if the study was requested
+     * * get main variant data: position, alleles, filter...
+     * * if there are empty alleles, get them from the vcf line
+     * * get the genotypes
+     * * add all (position, alleles, genotypes...) to a VariantContext for each study.
+     *
+     * @param variant
+     * @return
+     */
+    public Map<String, VariantContext> convertBiodataVariantToVariantContext(
+            Variant variant, Map<String, VariantSource> sources) 
+            throws IOException {
+
+        List<String> studyIds = new ArrayList<>(sources.keySet());
```

Comment:
Since the variable is only used once when invoking ```studyIds.contains()```, it is not necessary to "cast" to List.